tests/unit/test_structured_config.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import unittest
+from pathlib import Path
+
+import yaml
+from ops.testing import Harness
+
+from charm import KafkaCharm
+
+CONFIG = str(yaml.safe_load(Path("./config.yaml").read_text()))
+ACTIONS = str(yaml.safe_load(Path("./actions.yaml").read_text()))
+METADATA = str(yaml.safe_load(Path("./metadata.yaml").read_text()))
+
+logger = logging.getLogger(__name__)
+
+
+class TestStructuredConfig(unittest.TestCase):
```

Comment:
**(non-blocking) nitpick:** Personally I would like to keep all unit-tests using `pytest`. As it's a pain to change it's OK to not do now!

Reply:
https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Does that address what you mean?

Reply:
It worked! Thanks for the pointer!